### PR TITLE
feat/genericheader

### PR DESCRIPTION
**Pull request by MaximeHavez**
# Description
Generic Header

# Linked Issues
N/A

# Changes
Add generic header

# Screenshots
N/A

# Tests
How has it been tested ?

- [x] Manually
- [ ] Automated tests
- [ ] QA
- [ ] Other

# Additional information
no

### CODE REVIEW

File: yaki_mobile/lib/Authentification.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            flex: 6,
+            child: Header(
+              pictoIcon: 'assets/images/dots.svg',
+              pictoPath: 'assets/images/authent.svg',
+              headerTitle: tr('headerTitle'),
+              headerHint: tr('headerHint'),
+            )),
+        Expanded(flex: 7, child: Container())
+      ],
+    ));
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/Authentification.dart:23]: please format the code

File: yaki_mobile/lib/app.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import 'appRouter.dart';
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    context.setLocale(Locale('en', 'EN'));
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/app.dart:14]: do not force locale, remove this line

File: yaki_mobile/lib/appRouter.dart [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import 'Authentification.dart';
+
+final GoRouter appRouter = GoRouter(routes: <GoRoute>[
+  GoRoute(
+    path: '/',
+    builder: (context, state) => Authentification(),
+    routes: const [],
+  )
+]);
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/appRouter.dart:11]: format code

File: yaki_mobile/lib/main.dart [hunk 4]
```diff
@@ -1,115 +1,20 @@
...
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
-
-  final String title;
-
-  @override
-  State<MyHomePage> createState() => _MyHomePageState();
+import 'app.dart';
+import 'package:easy_localization/easy_localization.dart';
+void main() async{
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/main.dart:4]: format the code

File: yaki_mobile/lib/presentation/ui/shared/views/Header.dart [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  final String pictoPath;
+  final String headerTitle;
+  final String headerHint;
+
+  Header(
+      {super.key,
+      required this.pictoIcon,
+      required this.pictoPath,
+      required this.headerTitle,
+      required this.headerHint});
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/shared/views/Header.dart:18]: format code

File: yaki_mobile/lib/presentation/ui/shared/views/Header.dart [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+
+
+
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/shared/views/Header.dart:24]: format code

File: yaki_mobile/lib/presentation/ui/shared/views/Header.dart [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    return Container(
+      decoration: BoxDecoration(
+          borderRadius: const BorderRadius.only(
+              bottomLeft: Radius.circular(16),
+              bottomRight: Radius.circular(16)),
+          color: const Color.fromARGB(255, 245, 205, 61),
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/presentation/ui/shared/views/Header.dart:30]: add color to styles

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

File: yaki_mobile/lib/main.dart [hunk 8]
```diff
@@ -1,115 +1,22 @@
...
+
+void main() async{
+  WidgetsFlutterBinding.ensureInitialized();
+  await EasyLocalization.ensureInitialized();
+  runApp(
+      EasyLocalization(
+          supportedLocales:
+          const [
+            Locale('en', 'UK'),
+            Locale('fr', 'FR'),
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/main.dart:13]: can be just `en` and `fr`

same for translations files `en.json` and `fr.json`

File: yaki_mobile/lib/main.dart [hunk 9]
```diff
@@ -1,115 +1,22 @@
...
+          ],
+          path: 'assets/translations',
+          fallbackLocale: const Locale('en', 'UK'),
```
Comment (PiotrFLEURY) [inline yaki_mobile/lib/main.dart:16]: just `en`

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]

**Review by PiotrFLEURY — APPROVED** [review]

**Review by FlorianFlouquet — APPROVED** [review]

**Pull request merged**